modules/packages/Futures.chpl

```diff
@@ -99,11 +112,13 @@ module Futures {
 
     type retType;
 
-    var value: single retType;
     var valid: bool = false;
+    var value: retType;
+    var state: atomic bool;
```

Comment:
I think I understand the overhead of CPU atomics better than that of sync/single variables. So, it felt more like I was choosing to deal with a known quantity over an unknown one. I can see how a single variable would make sense here. If you think that'd be a better approach, I'm open to the idea of changing it.